test/lib/ansible_test/_data/sanity/code-smell/runtime-metadata.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+"""Schema validation of ansible-base's ansible_builtin_runtime.yml and collection's meta/runtime.yml"""
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import datetime
+import os
+import re
+import sys
+import yaml
+
+from voluptuous import Any, MultipleInvalid, PREVENT_EXTRA
+from voluptuous import Required, Schema, Invalid
+from voluptuous.humanize import humanize_error
+
+from ansible.module_utils.six import string_types
+
+
+def isodate(v):
+    """Validate a datetime.date or ISO 8601 date string."""
+    # datetime.date objects come from YAML dates, these are ok
+    if isinstance(v, datetime.date):
+        return v
+    # make sure we have a string
+    msg = 'Expected ISO 8601 date string (YYYY-MM-DD), or YAML date'
+    if not isinstance(v, string_types):
+        raise Invalid(msg)
+    try:
+        datetime.datetime.strptime(v, '%Y-%m-%d').date()
+    except ValueError:
+        raise Invalid(msg)
+    return v
+
+
+def main():
+    """Validate runtime"""
+    ansible_builtin_runtime = 'lib/ansible/config/ansible_builtin_runtime.yml'
+    collection_legacy_file = 'meta/routing.yml'
+    collection_runtime_file = 'meta/runtime.yml'
+    path = ''
+
+    # Check if the correct routing files can be found
+
+    if os.path.isfile(ansible_builtin_runtime):
```

Comment:
Instead of consulting the filesystem, accept the list of paths like other sanity tests and check for the file paths in that list.